### PR TITLE
Issue 3350 HttpMessage correct queryEquals method

### PR DESCRIPTION
**Pull request by kingthorin**
`queryEquals` correct comparison and add JavaDoc.

Addresses zaproxy/zaproxy#3350

### CODE REVIEW

**Comment by thc202** [comment]
Lets hold this, per comments in #3350.

File: src/org/parosproxy/paros/network/HttpMessage.java [hunk 1]
```diff
@@ -479,15 +480,27 @@ public boolean equalType(HttpMessage msg) {
 	    return result;
 	}
 	
+	/**
+	 * Compares the parameter names used in GET and POST messages. For POST
```
Comment (thc202) [inline src/org/parosproxy/paros/network/HttpMessage.java]: Could you add this explanation to `equalType` too? That's the method the users of the class will use/see (without checking the source).

Comment (kingthorin) [inline src/org/parosproxy/paros/network/HttpMessage.java]: Yup

**Comment by thc202** [comment]
Thanks!

**Pull request merged**